pages/index.js

```diff
@@ -286,37 +320,81 @@ const ShoppingList = () => {
     setDeletingItemId(null);
   };
 
+  // const toggleFormVisibility = () => {
+  //   setIsFormVisible((prev) => !prev);
+  // };
+
+  const handleCancel = (e) => {
+    e.preventDefault();
+    setIsFormVisible(false);
+    e.target.closest("form").reset();
+  };
```

Comment:
You don't need this function. Your button would also work if you gave it the attribute `type="reset"` and an `onClick` that only sets the `isFormVisible` state to `false`.